src/util.ts

```diff
@@ -35,3 +37,41 @@ export function normalize<T = {}, U = Function>(
 export function objectEntries<T>(obj: {[key: string]: T}): Array<[string, T]> {
   return Object.keys(obj).map(key => [key, obj[key]] as [string, T]);
 }
+
+/**
+ * URI encode the given string for generating signed URLs.
+ *
+ * This implementation differs from encodeURIComponent() in these ways:
+ *  - `! * ' ( )` characters are encoded;
+ *  - `/` is not encoded if encodeSlash is true.
+ * @param {string} uri The URI to encode.
+ * @param [boolean=false] encodeSlash If `true`, the "/" character is not encoded.
+ * @return {string} The encoded string.
+ */
+export function encodeURI(uri: string, encodeSlash: boolean): string {
+  // Encode using JavaScript's encodeURIComponent, excluding "/" if encodeSlash is
+  // `true`.
+  const encoded = uri
+    .split('/')
+    .map(encodeURIComponent)
+    .join(encodeSlash ? '%2F' : '/');
+
+  // Encode additional characters not encoded by encodeURIComponent.
```

Comment:
Maybe just follow this [example](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent) from MDN:

```js
function fixedEncodeURIComponent(str) {
  return encodeURIComponent(str).replace(/[!'()*]/g, function(c) {
    return '%' + c.charCodeAt(0).toString(16);
  });
}
```
same but more readable IMHO.

Reply:
fixed in #905